src/toPlaylists.js

```diff
@@ -0,0 +1,64 @@
+import { range } from './utils/list';
+import { getAttributes, shallowMerge } from './utils/object';
+import resolveUrl from './resolveUrl';
+
+export const segmentsFromTemplate = (attributes) => {
+  const startNumber = parseInt(attributes.startNumber, 10);
+  const duration = parseInt(attributes.duration, 10);
+  const timescale = parseInt(attributes.timescale, 10);
+
+  const segmentDuration = (duration / timescale);
+  const segmentCount = Math.round(attributes.sourceDuration / segmentDuration);
+
+  const indexes = range(segmentCount, startNumber);
+
+  return indexes.map(i => {
+    // TODO: deal with padding control characters
+    // $<token>...$
+    const uri = attributes.media.replace(/\$Number\$/gi, i)
+      .replace(/\$RepresentationID\$/gi, attributes.id);
+
+    return {
```

Comment:
This will need to have the initialization segment added as a property
`map: { uri, byteRange }` It's the only way right now that segment-loader uses to get the initialization segment detect if it needs to transmux from ts to mp4